Core/src/Definitions/Common.cpp

```diff
@@ -0,0 +1,33 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Definitions/Common.hpp"
+
+#include <ostream>
+
+namespace Acts {
+
+std::ostream& operator<<(std::ostream& os, MaterialUpdateStage matUpdate) {
+  switch (matUpdate) {
+    case MaterialUpdateStage::PreUpdate:
+      os << "PreUpdate (-1)";
+      break;
+    case MaterialUpdateStage::PostUpdate:
+      os << "PostUpdate (1)";
+      break;
+    case MaterialUpdateStage::FullUpdate:
+      os << "FullUpdate (0)";
+      break;
+    default:
```

Comment:
It might be useful to mark this code as unreachable to the compiler.

Reply:
That's a valid point. That wouldn't result in a compilation error with an unhandled value though right?

Reply:
It shouldn't, no. But thinking about it, marking it as unreachable might be a little heavy-handed. Sure, the compiler can optimize your code better that way, but it can also create undefined behaviour if someone managed to pass an invalid enum to your function... They probably shouldn't, but it's possible... Maybe we should just leave it like this.

Reply:
We try not to use these output functions in hot code paths anyway, so any potential optimization should have negligible effect.